src/App.tsx

```diff
@@ -1,35 +1,276 @@
+            </label>
+            <div className="flex gap-2 mt-1">
+              <Input
+                {...amountInput.register}
+                id={amountInputId}
+                className="flex-1"
+                placeholder="금액을 입력하세요"
+                allowNumberOnly
+                data-testid="lotto-amount-input"
+                maxLength={7}
+              />
+              <Button
+                type="submit"
+                disabled={amountInput.value.length === 0}
+                data-testid="lotto-purchase-button"
+              >
+                구매
+              </Button>
+            </div>
+            {amountInput.error && (
+              <p
+                className="mt-1 text-sm font-medium text-error"
+                data-testid="error-message"
+              >
+                {amountInput.error}
+              </p>
+            )}
+          </form>
+          <ul className="mt-8 flex flex-col gap-2">
+            {purchasedLotto.map((lotto, index) => (
+              <li
+                className="flex gap-1 border rounded-lg h-12 items-center px-3 justify-center"
+                key={index}
+              >
+                <LottoItem lotto={lotto} />
+              </li>
+            ))}
+          </ul>
+          <Button
+            className="mt-8 w-full"
+            onClick={showResult}
+            disabled={purchasedLotto.length === 0}
+            data-testid="show-result-button"
+          >
+            결과 확인
+          </Button>
+          {winningLotto !== null && <Result winningLotto={winningLotto} />}
+          <Button
+            className="mt-4 w-full"
+            onClick={onStartOverButtonClick}
+            data-testid="reset-button"
+          >
+            처음부터 다시하기
+          </Button>
+        </div>
+        <LottoHistory />
+        <LottoStatistics />
+      </main>
+    </div>
+  );
+};
+
+type LottoItemProps = {
+  lotto: Lotto;
+  bonusNumber?: number;
+};
+
+const LottoItem = ({ lotto, bonusNumber }: LottoItemProps) => {
+  return (
+    <div className="flex gap-1 items-center" data-testid="lotto-item">
+      {lotto.map((number, index) => (
+        <span
+          key={index}
+          className="border bg-gray-400 flex justify-center items-center rounded-full size-8 text-sm text-white"
+          data-testid="lotto-number"
+        >
+          {number}
+        </span>
+      ))}
+      {bonusNumber && (
+        <span
+          className="border bg-yellow-400 flex justify-center items-center rounded-full size-8 text-sm text-white"
+          data-testid="lotto-bonus-number"
+        >
+          {bonusNumber}
+        </span>
+      )}
+    </div>
+  );
+};
+
+type ResultProps = {
+  winningLotto: WinningLotto;
+};
+
+const labelByPrize: Record<LottoPrize, string> = {
+  1: "1등",
+  2: "2등",
+  3: "3등",
+  4: "4등",
+  5: "5등",
+  none: "꽝",
+};
+
+const Result = ({ winningLotto }: ResultProps) => {
+  const { purchasedLotto } = useLottoStore();
```

Comment:
p1: 이렇게 표현되는게 자연스러울 것 같아요
```suggestion
  const { purchasedLottos } = useLottoStore();
```